src/yachtzee.ts

```diff
@@ -0,0 +1,831 @@
+      }),
+      new DietEntry(`Extrovermectin™`, 0, 0, 0, 2, (n: number) => {
+        chew(n, $item`Extrovermectin™`);
+      }),
+      new DietEntry("synthesis", 0, 0, 0, 1, (n: number) => {
+        synthesize(n, $effect`Synthesis: Greed`);
+      }),
+      new DietEntry(`mojo filter`, 0, 0, 0, -1, (n: number) => {
+        use(n, $item`mojo filter`);
+      }),
+      new DietEntry(`beggin' cologne`, 0, 0, 0, 1, (n: number) => {
+        chew(n, $item`beggin' cologne`);
+      }),
+      new DietEntry(`stench jelly`, 0, 0, 0, 1, (n: number) => {
+        chew(n, $item`stench jelly`);
+      }),
+      new DietEntry(`jumping horseradish`, 0, 1, 0, 0, (n: number) => {
+        eat(n, $item`jumping horseradish`);
+      }),
+    ];
+    if (action) this.dietArray.forEach((entry) => (entry.action = action));
+  }
+
+  public setDietEntry(
+    name: string,
+    qty?: number,
+    action?: (n: number, name?: string) => void
+  ): void {
+    this.dietArray.forEach((entry) => {
+      if (entry.name === name) {
+        if (qty) entry.quantity = qty;
+        if (action) entry.action = action;
+      }
+    });
+  }
+
+  public resetDietPref(): void {
+    this.originalPref = "";
+    this.pref = "";
+  }
+
+  public addToPref(n: number, name?: string): void {
+    if (!name) throw "Diet pref must have a name";
+    for (let i = 0; i < n; i++) {
+      this.pref = this.pref.concat(name ?? "").concat(",");
+    }
+  }
+
+  public setDietPref(): void {
+    set("_garboYachtzeeChainDiet", this.originalPref.concat(this.pref));
+  }
+}
+
+function splitDietEntry(entry: DietEntry<void>): Array<DietEntry<void>> {
+  const entries = new Array<DietEntry<void>>();
+  for (let i = 0; i < entry.quantity; i++) {
+    entries.push(
+      new DietEntry(entry.name, 1, entry.fullness, entry.drunkenness, entry.spleen, entry.action)
+    );
+  }
+  return entries;
+}
+
+function combineDietEntries(left: DietEntry<void>, right: DietEntry<void>): DietEntry<void> {
+  return new DietEntry(
+    left.name,
+    left.quantity + right.quantity,
+    left.fullness,
+    left.drunkenness,
+    left.spleen,
+    left.action
+  );
+}
+
+function castOde(turns: number): boolean {
+  if (!have($skill`The Ode to Booze`)) return false;
+  for (const song of getActiveSongs()) {
+    const slot = Mood.defaultOptions.songSlots.find((slot) => slot.includes(song));
+    if (
+      !slot &&
+      song !== $effect`Ode to Booze` &&
+      song !== $effect`Polka of Plenty` &&
+      song !== $effect`Chorale of Companionship` &&
+      song !== $effect`The Ballad of Richie Thingfinder`
+    ) {
+      cliExecute(`shrug ${song}`);
+    }
+  }
+  // Shrug default Mood songs
+  cliExecute(`shrug ur-kel`);
+  cliExecute(`shrug phat loot`);
+
+  while (haveEffect($effect`Ode to Booze`) < turns) {
+    useSkill($skill`The Ode to Booze`);
+  }
+  return true;
+}
+
+function executeNextDietStep(stopBeforeJellies?: boolean): void {
+  if (property.getBoolean("stenchJellyUsed")) return;
```

Comment:
```suggestion
  if (get("stenchJellyUsed", false)) return;
```

Is there a reason this isnt' a _ property?

We should maybe add native mafia tracking for this

Reply:
Oh huh so Clara's Bell probably lasts across RO, but stench jelly doesn't. Makes sense to make it a _ property then.
I do think it's probably helpful to have it as a Mafia property, but it sounds like quite a bit of work differentiating between NCs that use a jelly charge and those that don't. Also, jelly charges can be stacked, so it should probably be a numeric value instead of a boolean in that case.